Pull request v6

lb3361 opened the pull request:
This is a pull request for ROMv6 which will be updated in light of the comments and concerns that will arise.

The native code of ROMv6 is 99.9% identical to current dev rom (one exception documented below) which has already been extensively tested.  Compared to ROMv5a, it brings some bug fixes, a few useful system calls, and a far more usable support for all kinds of ram & io expansion boards like the one Marcel was developing.  Many of the application changes were already present in the dev rom (faster Mandelbrot, workable card boot, racer improvements).  This pull request proposes to change the application selection by replacing "Pictures" and "Tetronis" by at67's "Invader" and hans61's "gtmine".  This can of course be changed. My reasons for choosing this two programs was that (a) Invader is widely recognized as a tour-de-force that illustrates what can be done within the boundaries of the classic ROMs, and (b) GtMine is a version of the classic minesweeper that one finds in virtually all operating systems, so why not the Gigatron rom. 

The pull request is made of seven commits described in detail below:

- Commit `2ab4ac4` "_[Prepare gtemuAT67 for ROMv6](https://github.com/kervinck/gigatron-rom/commit/2ab4ac418805b3a623edbb41b57acac48de478c7)_" includes minimal changes to ensure that gtemuat67 does not reject the new rom because of its new romTypeValue, changes to loader_config.ini to offer an easy way to load a ROMv6 in the current directory, and a small change that helps Macs to locate this ini file. 
- Commit `daa12d1` "_[Removed: Pictures, Tetronis. Added: Invader (at67)](https://github.com/kervinck/gigatron-rom/commit/d4a12d114e22287fc540499bbad0e7ab7aa2689a)_" adds an Apps subdirectories containing a pristine gt1 files for "Invader" and updated the main menu with this new application selection.
- Commit `8735f48` "_[Added: GtMine (hans61)](https://github.com/kervinck/gigatron-rom/commit/8735f48c0e0b5247c1adadaa3cc1630907c2877d)_" adds an Apps subdirectory containing Hans' GTMine program, as well as a main menu update.
- Commit `def6b66` "_[updated: Credits (smoothScroll)](https://github.com/kervinck/gigatron-rom/commit/def6b662bff3a6b9804efba6810740ebaeb64af1)_" updates the credits application. My goal was to make sure to list the contributions of at67 and hans61. Alas the old Credit screen was already cluttered. The new smooth scrolling interface provides an easy framework to reproduce the original credits and insert new credit lines for Invader and GtMine. This is easy to test because the program is not interactive: if it goes for 30 minutes without crashing, it goes forever.
- Commit `e6da9fa` "_[Novatron patch (also useful for certain ram & io expansion boards)](https://github.com/kervinck/gigatron-rom/commit/e6da9fa7dc5d246dcc2bf4185975cd566ac69533)_" is the only change to the dev.rom native code. The main reason to include this patch is to help Hans' [recreation of Marcel's original expansion board design with just 74' series chips](https://forum.gigatron.io/viewtopic.php?p=3694#p3694). The main drawback is that it reduces the number of SPI channels on Marcel's board to only two usable channels. Although Marcel's design offered 4 channels, only one is actually usable with common SPI devices. The patch is implemented using the conditional compilation support that was added to `asm.py` a while ago. It can be disabled by simply adding option `-DWITH_NOVATRON_PATCH=0` to the command line.
- Commit `7b96409` "_[Renamed DEVROM SYScalls as v6. Updated interface.json.](https://github.com/kervinck/gigatron-rom/commit/7b96409c36ffa0918925619d960cf62d6fea20a9)_" renames all the new syscalls with suffix `_DEVROM` to suffix` _v6`. This requires changes in `dev.asm.py`, in `interface.json`, and in all the dev rom applications that has been modified to use these new sys calls. This probably is the most error prone change necessary for a mainline rom release.  This commit also changes `RomTypeValue` to `romTypeValue_ROMv6` making the dev rom a prototype of ROMv6.
- The final commit `2009244` "_[Versionned apps and created ROMv6](https://github.com/kervinck/gigatron-rom/commit/20092441513c67e8d8bb3a75bf02280b5957c294)_" gives version numbers to all ROMv6 applications  versioned one (see [Docs/Versioning.txt](https://github.com/kervinck/gigatron-rom/blob/master/Docs/Versioning.txt)), copies `dev.asm.py` to `ROMv6.asm.py`, and restores the DEVROM romTypeValue in `dev.asm.py` making it again a DEVROM that is otherwise similar to ROMv6.  Binaries for ROMv6 are then included.

at67 commented:
1) gtemuAT67 pull request is in the wrong repo, I have already added the changes to the correct repo.
2) As previously stated Invader is not available for any other ROM's except ROMvX0, no at67 applications are available for any ROM's other than ROMvX0.

Fix 1) and 2) by removing them from the pull request and I will accept the pull request.

lb3361 commented on Apps/Mandelbrot/Mandelbrot.gcl:
```diff
@@ -73,7 +73,7 @@ $1ff [do i= A= B= 1<< SquareTable+ ptr= SlowMulShift7! ptr: i 1- if>=0loop]
 
 {-----------------------------------------------------------------------+
-|} $0300 \vLR: ret{          RAM page 3                                      |
+|} $0300 \vLR: [ret]{          RAM page 3                                      |
```
Writing [ret] instead of ret prevents compilegcl.py from creating a stub in the screen area.

lb3361 commented:
The pull request has been updated:
- the gtemuat67 patch has been removed
- the Invader application has been removed. Pictures has been reinstated.

Here is a breakthrough of all the commits:

- [Removed Tetronis. Added GtMine](https://github.com/kervinck/gigatron-rom/pull/249/commits/a2ed5d426a89fc5a28fe9c081439d357125ab2b5) - A new Apps subdirectory has been created for GtMine with a makefile. The MainMenu has been updated.
- [Faster Mandelbrot](https://github.com/kervinck/gigatron-rom/pull/249/commits/07c7280c67da15649b401e7f3f0e3ca914dd58a9) - This gains another 20% speed on Mandelbrot. This is an easy thing to test since it suffices to let it run for an extended period.
- [new Credits (scroll style)](https://github.com/kervinck/gigatron-rom/pull/249/commits/e522d1bebef3af97f9434dde0e5ce2583d84c4d8) - This is the updated credit application with a makefile. 
- [Added define -DWITH_SPI_BITS to control the SPI bitmask.](https://github.com/kervinck/gigatron-rom/pull/249/commits/d7e2c21c745a2b10275e5b0a7645706f0abc7eec) - This patch provides a compile-time option to fine tune the number of input SPI bits. This is mostly in the interest of providing a good experience with hans61 recreation of Marcel's original expansion board (only 7400 chips, no gal).  This is why the default value is 2.
- [Renamed DEVROM SYScalls as v6. Updated interface.json.](https://github.com/kervinck/gigatron-rom/pull/249/commits/c06fbf888933e2c1682846f251ebc79c602ffd27) This patch renames all the _DEVROM sys calls to _v6 and also updates interface.json and dev-interface.json accordingly. A couple applications (cardboot and tiny basic) have to be updated for the new names.
- [Versionned all apps (dev.rom is now like romv6)](https://github.com/kervinck/gigatron-rom/pull/249/commits/46ad8c27517865ca0b52cf5ad16eb7e068906bda) - Following Marcel's [directions](https://github.com/kervinck/gigatron-rom/blob/master/Docs/Versioning.txt) this patch gives a version number to all files involved in ROMv6. At this point dev.rom is equivalent to the future ROMv6.
- [Created ROMv6 and new dev.rom](https://github.com/kervinck/gigatron-rom/pull/249/commits/fc24c81542f3b999c052d0c43a07015573a19dde) - And this final patch renames dev.rom as ROMv6 and creates a new dev.rom that is similar to ROMv6 except for having `RomTypeValue_DEVROM`.
